File: src/main/java/com/example/demo/config/SecurityConfig.java
package com.example.demo.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpMethod;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
import org.springframework.security.web.SecurityFilterChain;

@Configuration
@EnableWebSecurity
public class SecurityConfig {

    @Bean
    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
        http
                .authorizeHttpRequests(authz -> authz  // Usar authorizeHttpRequests en lugar de authorizeRequests
                        .requestMatchers(HttpMethod.POST, "/api/users").permitAll() // Permitir acceso sin autenticación a este endpoint
                        .anyRequest().authenticated() // Requiere autenticación para cualquier otra solicitud
                );


        http.csrf().disable();
        return http.build();
    }
}

File: src/main/java/com/example/demo/security/SecurityConfig.kt
package com.example.demo.security

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.web.SecurityFilterChain

@Configuration
@EnableWebSecurity
open class SecurityConfig {

    @Bean
    open fun securityFilterChain(http: HttpSecurity): SecurityFilterChain {
        // Si quieres configurar el acceso, descomenta y ajusta la línea:
        // http.authorizeHttpRequests { authz -> authz.requestMatchers(HttpMethod.POST, "/api/users").permitAll() }
        //     .anyRequest().authenticated()

        http.authorizeHttpRequests { authz -> authz.anyRequest().permitAll() }
            .csrf { it.disable() }
        return http.build()
    }

    @Bean
    open fun authenticationManager(authConfig: AuthenticationConfiguration): AuthenticationManager {
        return authConfig.authenticationManager
    }

    @Bean
    open fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }
}